source/server/connection_handler_impl.cc

```diff
@@ -288,5 +311,56 @@ ListenerStats ConnectionHandlerImpl::generateStats(Stats::Scope& scope) {
   return {ALL_LISTENER_STATS(POOL_COUNTER(scope), POOL_GAUGE(scope), POOL_HISTOGRAM(scope))};
 }
 
+ConnectionHandlerImpl::ActiveUdpListener::ActiveUdpListener(ConnectionHandlerImpl& parent,
+                                                            Network::ListenerConfig& config)
+    : ActiveUdpListener(parent, parent.dispatcher_.createUdpListener(config.socket(), *this),
+                        config) {}
+
+ConnectionHandlerImpl::ActiveUdpListener::ActiveUdpListener(ConnectionHandlerImpl& parent,
+                                                            Network::ListenerPtr&& listener,
+                                                            Network::ListenerConfig& config)
+    : ConnectionHandlerImpl::ActiveListenerBase(parent, std::move(listener), config),
+      udp_listener_(dynamic_cast<Network::UdpListener*>(listener_.get())) {
+  // TODO(sumukhs): Try to avoid dynamic_cast by coming up with a better interface design
+  ASSERT(udp_listener_ != nullptr, "");
+
+  // Create the filter chain on creating a new udp listener
+  config_.filterChainFactory().createUdpListenerFilterChain(*this, *this);
+
+  // If filter chain is empty, fail the creation of the listener
+  if (read_filters_.empty()) {
+    throw Network::CreateListenerException(fmt::format(
+        "Cannot create listener as no read filters registered for the udp listener: {} ",
+        config_.name()));
+  }
+}
+
+void ConnectionHandlerImpl::ActiveUdpListener::onData(Network::UdpRecvData& data) {
+  ASSERT(read_filters_.size() == 1, "Udp listener only supports 1 filter");
```

Comment:
Where do you actually check this? Can you make sure we catch this at config load time and fail it (and have a test)?